orbit/pkg/execuser/execuser_windows.go

```diff
@@ -230,3 +238,62 @@ func startProcessAsCurrentUser(appPath, cmdLine, workDir string) error {
 
 	return nil
 }
+
+// setupSyncChannel creates the synchronization channel through which child process will be
+// signalled for termination. Code is using kernel named object as the sync primitive.
+// https://learn.microsoft.com/en-us/windows/win32/sync/using-event-objects
+func setupSyncChannel(channelId string) error {
+	if channelId == "" {
+		return errors.New("communication channel name should not be empty")
+	}
+
+	// converting go string to UTF16 windows compatible string
+	targetChannel := "Global\\comm-" + channelId
+	ev, err := windows.UTF16PtrFromString(targetChannel)
+	if (err != nil) && (err != windows.ERROR_SUCCESS) {
+		return fmt.Errorf("there was a problem generating UTF16 string: %w", err)
+	}
+
+	// checking first if channel is already present through the OpenEvent API
+	// OpenEvent API opens a named event object from the kernel object manager
+	// https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-openeventw
+	ha, err := windows.OpenEvent(windows.EVENT_ALL_ACCESS, false, ev)
+	if (ha != windows.InvalidHandle) && (err == nil) {
+		windows.CloseHandle(ha) // closing the handle to avoid handle leaks
```

Comment:
Given the comment when creating the handle (`windows.CreateEvent`) below:
```
// CreateEvent Api creates the named event object on the kernel object manager
// handle is not closed on purpose, the lifetime of the event object
// is going to be bound to the lifetime of the service
```
curious how can this happen?

Reply:
This happens because kernel objects on Windows are reference counted, which means that they are going to exist if there is at least one reference to them (details [here](https://learn.microsoft.com/en-us/windows-hardware/drivers/kernel/life-cycle-of-an-object)). In this particular example, Orbit is creating a named event kernel open through the `CreateEvent` call. This call returns a windows handle to the process which implicitly increments the object's reference count on the kernel object. Then if the code just calls `windows.CloseHandle` on this handle, the reference count will be decremented, and the object will be freed/removed on the kernel side. We don't want that as we need the object to be there and listened to by fleet-desktop, so in this scenario the code is just not closing the handle and leaving the named event kernel object to live for the lifetime of the process (the windows kernel is going to close any handle while freeing the resources associated to the process). The screenshot below shows how the event handle will look like on the orbit process handles table

![image](https://user-images.githubusercontent.com/17311749/195095364-9cb8f3aa-2a0a-4da5-be99-2c4e24e48dac.png)